dask_planner/src/parser.rs

```diff
@@ -375,4 +437,48 @@ impl<'a> DaskParser<'a> {
             schema_name,
         })))
     }
+
+    /// Retrieves the table_schema and table_name from a `TableFactor` instance
+    fn elements_from_tablefactor(tbl_factor: &TableFactor) -> (String, String) {
+        match tbl_factor {
+            TableFactor::Table {
+                name,
+                alias: _,
+                args: _,
+                with_hints: _,
+            } => {
+                let identities: Vec<String> = name.0.iter().map(|f| f.value.clone()).collect();
+
+                assert!(identities.len() <= 2 && !identities.is_empty());
+
+                if identities.len() == 1 {
+                    ("".to_string(), identities[0].clone())
+                } else if identities.len() == 2 {
+                    (identities[0].clone(), identities[1].clone())
+                } else {
+                    ("".to_string(), "".to_string())
+                }
```

Comment:
The `else` seems to be unreachable here, and I think this is a better fit for a `match` statement:


```suggestion
                match identities.len() {
                    1 => ("".to_string(), identities[0].clone()),
                    2 => (identities[0].clone(), identities[1].clone()),
                    _ => panic!("TableFactor name only supports 1 or 2 elements")
                }
```